app/src/org/commcare/android/util/SigningUtil.java

```diff
@@ -0,0 +1,220 @@
+package org.commcare.android.util;
+
+import org.commcare.android.logic.GlobalConstants;
+import org.spongycastle.jce.provider.BouncyCastleProvider;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+import java.net.URL;
+import java.security.InvalidKeyException;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.Signature;
+import java.security.SignatureException;
+import java.security.spec.X509EncodedKeySpec;
+
+/**
+ * A set of helper methods for verifying whether a message was genuinely sent from HQ. Currently we
+ * expcect the SMS in the format [COMMCARE APP - DO NOT DELETE] {BASE64-Encoded-Message} EG
+ *
+ * [COMMCARE APP - DO NOT DELETE] Y2NhcHA6IGh0dHA6Ly9iaXQubHkvMU5JMUl6MyBzaWduYXR1cmU6IEvECygFUhiUH
+ * 3TRjC0lClQrpLR7lG//IpDYpRH7ComtZRjTirteXmPyM9fRgbPZ9K6jG9zEms9WQj55Uo7jTujKNYThjU8rJJmWLouJBr/Yn
+ * WobEupwzn6DP2FavPF1YLPbp0ZctOfymW3m4j3VZ0lR2dMOjmInMSBiInqICKid
```

Comment:
we should update this comment since now the format of the sms is:
`[commcare app - do not delete] <url>`

once you follow the url, you get the base64-encoded message which is decoded the same way described here below